### DISCUSSION TITLE
[office-js] xxxEventArgs event.type are inconsistent within the lib (Excel.EventType)

### DISCUSSION THREAD

**Discussion by danielheuer** [DefinitelyTyped/DefinitelyTyped#71313] [Issues with a @types package]
xxxEventArgs are inconsistent within the whole the lib to get correct narrowing through typescript in switch event.type including // https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking

Either all type must be type "string" or "xxxx" or Excel.EventType (enum). I personaly prefer to use the enum type, but i know, this is maybe a performance issue.

Example
interface TableChangedEventArgs  {
...
type: "TableChanged" or Excel.EventType.tableChanged or string
}

On this occasion, i found, that a lot of types in the lib (i investigated the Excel-Namespace) are "double" defined. One with the appropriate enum and als the constants. Example here :
interface CommentDeletedEventArgs {
...
        source: Excel.EventSource | "Local" | "Remote";
    }

this also results in errors when do a exhaustiveness-checking within typescipt

tsc -v: Version 5.6.2
"name":"@types/office-js",
  "version":"1.0.449",
  "description":"TypeScript definitions for office-js",

[a comment by an automated account is not shown]

**Reply by danielheuer** [reply]
example Excel.TableChangedEventArgs in [office-js]

**Comment by ElizabethSamuel-MSFT** [comment]
@danielheuer Thanks for asking about this. This came about because of the support levels of some of the Office.js enums. Per your example, Excel.EventType lists a number of event types available in Excel. However, TableChangedEventArgs only supports the TableChanged event type. The general intention is to allow you to use the enum or string value and have compile-time type checking instead of only at runtime, especially for situations where an API only supports a subset of the available values.

If I haven't addressed your concern, do let us know.

Thanks.

**Reply by danielheuer** [reply]
i agree, but there are some type-types in some *EventArgs, that are defined as string only and not as "xxx" or Excel.EventType.xxx respectivly. 
So, i opened a issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/71314.

see example in index.d.ts (
    interface WorksheetNameChangedEventArgs {
...
        /**
         * Gets the type of the event.
         *
         * @remarks
         * [Api set: ExcelApi 1.17]
         */
        type: string;
...
    }
...
    interface WorksheetVisibilityChangedEventArgs {
...
        /**
         * Gets the type of the event.
         *
         * @remarks
         * [Api set: ExcelApi 1.17]
         */
        type: string;
        /**
...
    }

**Reply by ElizabethSamuel-MSFT** [reply]
I see. In that case, let's consider this discussion closed and we'll track using the issue.
